packages/gradient_boosting_model/gradient_boosting_model/processing/preprocessors.py

```diff
@@ -0,0 +1,64 @@
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+
+
+class SklearnTransformerWrapper(BaseEstimator, TransformerMixin):
+    """Categorical data missing value imputer."""
+
+    def __init__(self, variables=None, transformer=None):
+
+        if not isinstance(variables, list):
+            self.variables = [variables]
+        else:
+            self.variables = variables
+
+        self.transformer = transformer
+
+    def fit(self, X: pd.DataFrame, y: pd.Series = None):
+        """Fit statement to accomodate the sklearn pipeline."""
+        self.transformer.fit(X[self.variables])
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """Apply the transforms to the dataframe."""
+        X = X.copy()
+        X[self.variables] = self.transformer.transform(X[self.variables])
+        return X
+
+
+class TemporalVariableEstimator(BaseEstimator, TransformerMixin):
+    """Temporal variable calculator."""
```

Comment:
Replace by:
"Calculates the time difference between 2 temporal variables"

Reply:
updated